sql/catalyst/src/main/java/org/apache/spark/sql/connector/join/JoinColumn.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.join;
+
+import org.apache.spark.annotation.Evolving;
+import org.apache.spark.sql.connector.expressions.Expression;
+import org.apache.spark.sql.connector.expressions.NamedReference;
+
+/**
+ * Represents a column reference used in DSv2 Join pushdown.
+ *
+ * @since 4.0.0
+ */
+@Evolving
+public final class JoinColumn implements NamedReference {
+  public JoinColumn(String name, Boolean leftSideOfJoin) {
+    this.name = name;
+    this.leftSideOfJoin = leftSideOfJoin;
+  }
+
+  private String name;
```

Comment:
shall we also use `String[]` in case we want to support join condition with nested columns in the future?